File: kth_binary_search.java
import java.util.*;
class kth {
    public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);
        int arr[]={2,3,4,7,11};
    int mid;
    int start =1;
    int n= arr.length;
    int end=n-1;
    int ans=arr.length;
    int anser;
    System.out.print("enter the number which resemble the missing number count = ");
    int target = sc.nextInt();
    
      while(start<=end){
       mid = (start+end)/2;
        
        if(arr[mid]-mid-1>=target){
            ans = mid;
            end = mid -1;
          
        }
        
        else{
            start = mid +1;
        }
    }
    anser = ans + target;
    System.out.println("missing number = "+ anser);
        }
}

File: kth_binary_search.cpp

#include <iostream>
using namespace std;
int main() {
    int arr[]={2,3,4,7,11};
    
    int find;
    cout<<"enter the number which resemble the missing number count = ";
    cin>>find;
    int n = sizeof(arr)/sizeof(arr[0]);
    int mid;
    int start =0;
    int end=n-1;
    int ans= arr.size();
    
    while(start<=end){
        mid = (start+end)/2;
        
        if(arr[mid]-mid-1>=find){
            ans = mid;
            end = mid -1;
          
        }
        
        else{
            start = mid +1;
        }
    }
    
    cout<<"missing number = "<<ans+find;
    

    return 0;
}
